src/containers/blocks/AddURLBlock.js

```diff
@@ -0,0 +1,83 @@
+import React, {Component} from 'react';
+import {connect} from "react-redux";
+import {bindActionCreators} from 'redux';
+import {addUrlToTest, hideUrlInput, toggleUrlInput, removeUrlFromTest} from "../../actions"
+
+class AddURLBlock extends Component {
+  constructor(props){
+    super(props);
+
+    this.state = {
+      urlValue: ""
+    }
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.hideUrlInput();
+  }
+
+  handleAddUrl() {
+    this.props.toggleUrlInput();
+  }
+
+  handleInputChange(value) {
+    this.setState({
+      urlValue: value
+    });
+  }
+
+  handleUrlAdded() {
+    if(this.state.urlValue.length > 1) {
+      this.props.addUrlToTest(this.state.urlValue);
+      this.setState({
+        urlValue: ""
+      });
+      this.props.toggleUrlInput()
+    }
+  }
+
+  handleUrlDeletion(url) {
+    this.props.removeUrlFromTest(url);
+  }
+
+  generateListOfUrls() {
+    if(this.props.urls.length > 0) {
```

Comment:
Is there need to protect from this being undefined or empty array?

Reply:
Yes, if it's empty (and it is empty when you create a new test) then there is a `map is not a function` error.